gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ProcedureRequestMapper.java

```diff
@@ -0,0 +1,172 @@
+         * - context: references an encounter resource if it has been generated from the ehrComposition (NIAD-2025)
+         * - fallback to a default 'Unknown User' Practitioner if none are present in requester (NIAD-2026)
+         * - concatenate source practice org id to identifier URL (NIAD-2021)
+         */
+
+        var id = planStatement.getId().getRoot();
+        var identifier = getIdentifier(id);
+        var note = getNote(planStatement.getText());
+        var reasonCode = new CodeableConceptMapper().mapToCodeableConcept(planStatement.getCode());
+        var authoredOn = getAuthoredOn(planStatement.getAvailabilityTime(), ehrExtract, planStatement.getId());
+        var occurrence = getOccurrenceDate(planStatement.getEffectiveTime());
+        var agentReference = getAgentReference(planStatement.getParticipant(), ehrExtract, planStatement.getId());
+
+        return createProcedureRequest(id, identifier, note, reasonCode, authoredOn, occurrence, agentReference);
+    }
+
+    private Identifier getIdentifier(String id) {
+        Identifier identifier = new Identifier()
+                .setSystem(IDENTIFIER_SYSTEM) // TODO: concatenate source practice org id to URL
+                .setValue(id);
+        return identifier;
+    }
+
+    private Annotation getNote(String text) {
+        if (StringUtils.isNotEmpty(text)) {
+            var note = new Annotation();
+            return note.setText(text);
+        }
+
+        return null;
+    }
+
+    private Date getAuthoredOn(TS availabilityTime, RCMRMT030101UK04EhrExtract ehrExtract, II planStatementID) {
+        if (availabilityTime != null) {
+            return DateFormatUtil.parse(availabilityTime.getValue()).getValue();
+        } else {
+            var ehrComposition = EhrResourceExtractorUtil.extractEhrCompositionForPlanStatement(ehrExtract, planStatementID);
+            if (ehrComposition.getAvailabilityTime() != null) {
+                return DateFormatUtil.parse(ehrComposition.getAvailabilityTime().getValue()).getValue();
+            } else if (ehrExtract.getAvailabilityTime() != null) {
+                return DateFormatUtil.parse(ehrExtract.getAvailabilityTime().getValue()).getValue();
+            }
+        }
+
+        return null;
+    }
+
+    private DateTimeType getOccurrenceDate(IVLTS effectiveTime) {
+        if (effectiveTime != null && effectiveTime.getCenter() != null) {
+            return DateFormatUtil.parse(effectiveTime.getCenter().getValue());
+        }
+
+        return null;
+    }
+
+    private Reference getAgentReference(List<RCMRMT030101UK04Participant> participantList, RCMRMT030101UK04EhrExtract ehrExtract,
+                                        II planStatementID) {
+        Reference reference = new Reference();
+        var nonNullFlavorParticipants = participantList.stream()
+                .filter(participant -> isNotNullFlavour(participant))
```

Comment:
Intellij suggested that method reference can be used instead of lambda here: `.filter(this::isNotNullFlavour)`